Update to exclude GTX 16xx from autocast selection

ArDiouscuros opened the pull request:
Proposed changes to solve #526.
Probably related to pytorch issue #77955 https://github.com/pytorch/pytorch/issues/77955

mh-dm commented on ldm/dream/devices.py:
```diff
@@ -13,9 +13,12 @@ def choose_torch_device() -> str:
-    elif device_type == 'cpu':
-        return device_type,nullcontext
+        device_name = torch.cuda.get_device_name()
+        if 'GTX 16' in device_name:
```
Nice fix. Though to prevent potential name clashes in the future, can you check against a more specific 'GeForce GTX 1660'? (As for 1650, don't remember seeing reports but it could easily be a separate check)

lstein reviewed (commented):
Thanks for this. There is also bug report #588 that came in this morning regarding green images on the GTX 1650. Should this fix be generalized to the whole 16XX series?

mh-dm commented:
Thanks, I couldn't find a report on 1650 before. @ArDiouscuros could you please add an or 'GeForce GTX 1650'? Also, can you please squash all commits into one (ex: git rebase -i) and update branch (ex: git pull --rebase)?

lstein commented:
> Thanks, I couldn't find a report on 1650 before. @ArDiouscuros could you please add an or 'GeForce GTX 1650'? Also, can you please squash all commits into one (ex: git rebase -i) and update branch (ex: git pull --rebase)?

I don't know much about the lineage of these video cards, but maybe we need a regular expression match for all GTX 16\d\d cards?

mh-dm commented:
1630 was released 2 years after, not sure if it's actually same generation / same characteristics or just named like that. Also, it has only 4GB so if it can't use float16 it won't be usable. Do we have any reports from 1630 users? It's easy to add another or 'GeForce GTX 1630' if reports come in.

ArDiouscuros commented:
> It's easy to add another or 'GeForce GTX 1630' if reports come in.

Agree, that's right.

> squash all commits into one (ex: git rebase -i) and update branch

Squashed and rebased.

mh-dm commented:
> Squashed and rebased.

I still see 3 commits in this PR?

lstein reviewed (approved):
Let's do it!

lstein commented:
I'll do the squash.
